yoga/Yoga.c

```diff
@@ -941,6 +955,41 @@ static inline YGDirection YGNodeResolveDirection(const YGNodeRef node,
   }
 }
 
+static float YGBaseline(const YGNodeRef node,
+                        const YGFlexDirection crossAxis) {
+  if (node->baseline != NULL) {
+    const float baseline = node->baseline(node);
+    if (YGFloatIsUndefined(baseline)) {
+      return node->layout.measuredDimensions[dim[crossAxis]];
+    }
+    return baseline;
+  }
+
+  YGNodeRef baselineChild = NULL;
+  for (uint32_t i = 0; i < YGNodeGetChildCount(node); i++) {
+    const YGNodeRef child = YGNodeGetChild(node, i);
+    if (child->style.positionType == YGPositionTypeAbsolute || child->lineIndex > 0) {
+      continue;
+    }
+
+    if (YGNodeAlignItem(node, child) == YGAlignBaseline) {
```

Comment:
The baseline of a container is defined by its first baseline aligned child, but when there is no baseline aligned child then the container's baseline is defined by it's last child. This seems odd. Are you sure we have test cases covering edge cases here to make sure this is how it works on the web?